src/main/java/org/spongepowered/common/mixin/core/network/NetHandlerPlayServerMixin.java

```diff
@@ -811,12 +811,16 @@ public void processUseEntity(final CPacketUseEntity packetIn) {
         this.player.markPlayerActive();
 
         if (entity != null) {
-            final boolean flag = this.player.canEntityBeSeen(entity);
-            double d0 = 36.0D; // 6 blocks
-
-            if (!flag) {
-                d0 = 9.0D; // 1.5 blocks
-            }
+            // Sponge start - flag is not necessary, see below
+            // final boolean flag = this.player.canEntityBeSeen(entity);
+            // compatibility with Forge reachDistance attribute
+            double d0 = SpongeImplHooks.getEntityReachDistanceSq(this.player);
+
+            // this check introduces MC-107103
```

Comment:
I'm actually not really sure about adding this straight up. It feels to me that this was intended to work as a way to ensure "line of sight" for an interaction. The removal of this check means that I could, in theory, reach though a few blocks to interact with a switch on the other side, when I don't think that's intended.

I don't think this "fix" is correct and that the real bug is client side. I can see hack clients possibly making use of this to activate things though walls and such.

Reply:
This line of sight check is only applied to interactions with entities, not blocks.
If you look at the bug description on Mojang's tracker, they talk about this exact if-statement.

Imagine this scenario. You are using a mob farm, where you 1) need to click at mobs' legs, while mobs are standing above you by 1 block, 2) there's a block that blocks line of sight between you and mobs' eyes, 3) you need to stand back 4 blocks (for one reason or another).
What happens is when you do the click - interaction only happens client-side. E.g. if you name a mob in that situation - you'll get a "ghost empty slot" in place of the name tag and a mob that shows a name but doesn't have one.

Again, this check only works on entities, as blocks are handled in a separate method.

Reply:
It's the same principle though. Should this line of sight check be removed for these entity checks? I'm not convinced that it's a server bug, but a client one - the client is what is causing the ghosting.

Reply:
Well, the only thing line of sight check does is it halves the distance the player can reach that entity from.
And remember, it doesn't check if player sees entity's hitbox. It checks if player sees entity's eye point.

Reply:
I know what it does.

My point is that the bug seems to be client side to me (the client ignores the fact this check exists) rather than this check exists in the first place. If we were to "fix" this, it really should be checking to see if we can see the hitbox rather than removing the check altogether, the check is clearly there for a reason. This is why I'm not sure I think we should accept this.

Reply:
Well, I can only pull out one last argument: actual Minecraft sources for 1.16.
```java
   public void processUseEntity(CUseEntityPacket packetIn) {
      PacketThreadUtil.checkThreadAndEnqueue(packetIn, this, this.player.getServerWorld());
      ServerWorld serverworld = this.player.getServerWorld();
      Entity entity = packetIn.getEntityFromWorld(serverworld);
      this.player.markPlayerActive();
      this.player.setSneaking(packetIn.func_241792_e_());
      if (entity != null) {
         double d0 = 36.0D;
         if (this.player.getDistanceSq(entity) < 36.0D) {
            // do the rest
```
That `if` statement was so gone that they even forgot to remove the d0 variable. :) 
I guess, if we are worrying about hacked clients, there could be some sort of RayTrace check, but the if-removal at least won't make Sponge less secure than Vanilla is.